doc/languages-frameworks/rust.section.md

```diff
@@ -107,6 +107,53 @@ rustPlatform.buildRustPackage rec {
 }
 ```
 
+### Importing a `Cargo.lock` file
+
+Using `cargoSha256` or `cargoHash` is not pleasant when using
+`buildRustPackage` within a project, since it requires that the hash
+it updated after every change to `Cargo.lock`. Therefore,
+`buildRustPackage` also supports vendoring dependencies directly from
+a `Cargo.lock` file using the `cargoLock` argument. For example:
+
+```nix
+rustPlatform.buildRustPackage rec {
+  pname = "myproject";
+  version = "1.0.0";
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+  }
+
+  # ...
+}
+
+```
+
+This will retrieve the dependencies using fixed-output derivations from
+the specified lockfile.
+
+For dependencies that use a git source, the output hash of that
+dependency needs to be specified. For example:
+
+```nix
+rustPlatform.buildRustPackage rec {
+  pname = "myproject";
+  version = "1.0.0";
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+    outputHash = {
```

Comment:
Should this be called `outputHashes`?

Reply:
I thought a lot about this and couldn't really make up my mind. Changed to `outputHashes`.